frontend/packages/dev-console/src/components/topology/topology-utils.ts

```diff
@@ -318,5 +375,9 @@ export const removeTopologyResourceConnection = (
   const sourceObj = getResourceDeploymentObject(source);
   const targetObj = getResourceDeploymentObject(target);
 
+  if (target.operatorBackedService) {
+    return removeServiceBinding(sourceObj, targetObj);
+  }
+
   return removeResourceConnection(sourceObj, targetObj);
```

Comment:
You need to check which is the `edge.type` being deleted here.
Update the following inside `D3ForceDirectedRenderer` to also include the edge type. If `connects-to` do the old delete, if `service-binding` do the new delete.
`onRemove={() => onRemoveConnection(viewEdge.source.id, viewEdge.target.id)}`